src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -0,0 +1,223 @@
+import jewellery.inventory.dto.request.resource.ResourceQuantityRequestDto;
+import jewellery.inventory.dto.response.ProductResponseDto;
+import jewellery.inventory.exception.not_found.*;
+import jewellery.inventory.exception.product.ProductIsContentException;
+import jewellery.inventory.exception.product.ProductIsSoldException;
+import jewellery.inventory.mapper.ProductMapper;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.ResourceInUser;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Resource;
+import jewellery.inventory.model.resource.ResourceInProduct;
+import jewellery.inventory.repository.*;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+
+  private final ProductRepository productRepository;
+  private final UserRepository userRepository;
+  private final ResourceInUserRepository resourceInUserRepository;
+  private final ResourceInUserService resourceInUserService;
+  private final ProductMapper productMapper;
+
+  @Transactional
+  public ProductResponseDto createProduct(ProductRequestDto productRequestDto) {
+    User owner = getUser(productRequestDto.getOwnerId());
+    Product product = createProductWithoutResourcesAndProducts(productRequestDto, owner);
+    addProductsContentToProduct(productRequestDto, product);
+    addResourcesToProduct(productRequestDto, owner, product);
+    return productMapper.mapToProductResponseDto(product);
+  }
+
+  private User getUser(UUID userId) {
+    return userRepository.findById(userId).orElseThrow(() -> new UserNotFoundException(userId));
+  }
+
+  private Product createProductWithoutResourcesAndProducts(
+      ProductRequestDto productRequestDto, User user) {
+    Product product = getProduct(productRequestDto, user);
+    productRepository.save(product);
+    return product;
+  }
+
+  private Product getProduct(ProductRequestDto productRequestDto, User user) {
+    Product product = new Product();
+    product.setOwner(user);
+    product.setAuthors(getAuthors(productRequestDto));
+    product.setSold(false);
+    product.setDescription(productRequestDto.getDescription());
+    product.setSalePrice(productRequestDto.getSalePrice());
+    product.setProductionNumber(productRequestDto.getProductionNumber());
+    product.setCatalogNumber(productRequestDto.getCatalogNumber());
+    return product;
+  }
+
+  private List<User> getAuthors(ProductRequestDto productRequestDto) {
+    List<UUID> authorsIds = productRequestDto.getAuthors();
+    List<User> authors = new ArrayList<>();
+    authorsIds.forEach(
+        id -> {
+          User author =
+              userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+          authors.add(author);
+        });
+    return authors;
```

Comment:
The solution is good as it is 👍 . Just recommending a "one-liner" in functional style.
```suggestion
    return productRequestDto.getAuthors().stream()
        .map(
            authorId ->
                userRepository
                    .findById(authorId)
                    .orElseThrow(() -> new UserNotFoundException(authorId)))
        .toList();
```

Reply:
Done!